pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -197,23 +231,23 @@ func TestDesiredRouterDeployment(t *testing.T) {
 			},
 		},
 	}
-
-	proxyNeeded, err := IsProxyProtocolNeeded(ci, infraConfig.Status.PlatformStatus)
-	if err != nil {
-		t.Errorf("failed to determine infrastructure platform status for ingresscontroller %s/%s: %v", ci.Namespace, ci.Name, err)
-	}
-	deployment, err := desiredRouterDeployment(ci, ingressControllerImage, ingressConfig, apiConfig, networkConfig, proxyNeeded, false, nil, false)
+	proxyNeeded, err := IsProxyProtocolNeeded(ic, infraConfig.Status.PlatformStatus)
 	if err != nil {
-		t.Errorf("invalid router Deployment: %v", err)
+		t.Errorf("failed to determine infrastructure platform status for ingresscontroller %s/%s: %v", ic.Namespace, ic.Name, err)
```

Comment:
I was going to suggest adding `t.Helper()` at the start of `getRouterDeploymentComponents`, but maybe it makes sense not to:

> Helper marks the calling function as a test helper function. When printing file and line information, that function will be skipped. Helper may be called simultaneously from multiple goroutines. 

By omitting `t.Helper()`, an error here will report the problem as an error in `getRouterDeploymentComponents` rather than its caller, and maybe that makes sense to do?

Reply:
No, I'll add the `t.Helper()` to `getRouterDeploymentComponents`, because it's better to know which caller had the issue, since the caller provides identifying information.